app/ts/Legend/LegendWrapper/LegendWrapper.tsx

```diff
@@ -0,0 +1,127 @@
+import {Dialog, DialogActions, DialogContent, Tooltip} from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import DialogTitleWithCross from 'app/ts/DialogTitleWithCross/DialogTitleWithCross';
+import createEnterHandler from 'app/ts/util/createEnterHandler';
+import DisplayErrors from 'app/ts/util/DisplayErrors';
+import {SubproblemContext} from 'app/ts/Workspace/SubproblemContext/SubproblemContext';
+import React, {ChangeEvent, useContext, useEffect, useState} from 'react';
+import {LegendContext} from '../LegendContext';
+import LegendButtons from './LegendButtons/LegendButtons';
+import LegendTable from './LegendTable/LegendTable';
+import LegendTooltip from './LegendTooltip/LegendTooltip';
+import {initLegend} from './LegendUtil';
+
+export default function LegendWrapper({
+  children
+}: {
+  children: any;
+}): JSX.Element {
+  const {canEdit, legend, saveLegend} = useContext(LegendContext);
+  const {filteredAlternatives} = useContext(SubproblemContext);
+
+  const [isDialogOpen, setDialogOpen] = useState<boolean>(false);
+  const [error, setError] = useState<string>();
+  const [isButtonPressed, setIsButtonPressed] = useState<boolean>(false);
+  const [newTitles, setNewTitles] = useState<Record<string, string>>(
+    initLegend(legend, filteredAlternatives)
+  );
+
+  useEffect(() => {
+    if (isDialogOpen) {
+      setIsButtonPressed(false);
+    }
+    setNewTitles(initLegend(legend, filteredAlternatives));
+  }, [isDialogOpen]);
+
+  const handleKey = createEnterHandler(handleLegendSave, isDisabled);
+
+  function openDialog(event: React.MouseEvent<HTMLButtonElement>): void {
+    setDialogOpen(true);
+  }
+
+  function closeDialog(): void {
+    setDialogOpen(false);
+  }
+
+  function handleLegendChange(
+    alternativeId: string,
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): void {
+    const newTitle = event.target.value;
+    setNewTitles({...newTitles, [alternativeId]: newTitle});
```

Comment:
isn't it nice how advancing understanding & language features lets one write lines like this :)